packages/eas-cli/src/commands/branch/publish.ts

```diff
@@ -83,12 +135,79 @@ export default class BranchPublish extends Command {
       if (jsonFlag) {
         throw new Error(validationMessage);
       }
-      ({ name } = await promptAsync({
-        type: 'text',
-        name: 'name',
-        message: 'Please enter the name of the branch to publish on:',
-        validate: value => (value ? true : validationMessage),
-      }));
+
+      const branches = await listBranchesAsync({ fullName: `@${accountName}/${slug}` });
+      name = await selectAsync<string>(
+        'which branch would you like to publish on?',
+        branches.map(branch => {
+          return {
+            title: `${branch.name} ${chalk.grey(
+              `- current update: ${formatUpdate(branch.updates[0])}`
+            )}`,
+            value: branch.name,
+          };
+        })
+      );
+    }
```

Comment:
This prompt looks like:
<img width="1329" alt="Screen Shot 2021-03-09 at 8 13 47 AM" src="https://user-images.githubusercontent.com/1220444/110501787-61994980-80af-11eb-8894-b3f714245ebb.png">